datafusion/core/tests/dataframe/mod.rs

```diff
@@ -2194,3 +2194,44 @@ async fn write_parquet_results() -> Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn test() -> Result<()> {
+    let schema = Arc::new(Schema::new(vec![Field::new("a", DataType::Int64, true)]));
+
+    let batch = RecordBatch::try_new(
+        schema.clone(),
+        vec![Arc::new(Int64Array::from(vec![Some(1), Some(3), Some(5)])) as ArrayRef],
+    )?;
+
+    let table = MemTable::try_new(schema, vec![vec![batch]])?;
+
+    let config =
+        SessionConfig::from_string_hash_map(std::collections::HashMap::from([(
+            "datafusion.sql_parser.dialect".to_owned(),
+            "postgresql".to_owned(),
+        )]))?;
+    let ctx = SessionContext::new_with_config(config);
+
+    ctx.register_table("test", Arc::new(table))?;
+
+    let sql = "SELECT MIN(a) FILTER (WHERE a > 1) AS x FROM test";
```

Comment:
I'm not sure what is this test for 🤔 

Is it possible to move to slt?

```
statement ok
set datafusion.sql_parser.dialect = 'Postgres';

statement ok
create table t (a float) as values (1), (2), (3);

query TT
explain select min(a) filter (where a > 1) as x from t;
----
logical_plan
01)Projection: MIN(t.a) FILTER (WHERE t.a > Int64(1)) AS x
02)--Aggregate: groupBy=[[]], aggr=[[MIN(t.a) FILTER (WHERE t.a > Float32(1)) AS MIN(t.a) FILTER (WHERE t.a > Int64(1))]]
03)----TableScan: t projection=[a]
physical_plan
01)ProjectionExec: expr=[MIN(t.a) FILTER (WHERE t.a > Int64(1))@0 as x]
02)--AggregateExec: mode=Single, gby=[], aggr=[MIN(t.a) FILTER (WHERE t.a > Int64(1))]
03)----MemoryExec: partitions=1, partition_sizes=[1]

statement ok
drop table t;
```

Reply:
Sure, I can do that.

Reply:
> I'm not sure what is this test for 🤔
> 
> Is it possible to move to slt?
> 
> ```
> statement ok
> set datafusion.sql_parser.dialect = 'Postgres';
> 
> statement ok
> create table t (a float) as values (1), (2), (3);
> 
> query TT
> explain select min(a) filter (where a > 1) as x from t;
> ----
> logical_plan
> 01)Projection: MIN(t.a) FILTER (WHERE t.a > Int64(1)) AS x
> 02)--Aggregate: groupBy=[[]], aggr=[[MIN(t.a) FILTER (WHERE t.a > Float32(1)) AS MIN(t.a) FILTER (WHERE t.a > Int64(1))]]
> 03)----TableScan: t projection=[a]
> physical_plan
> 01)ProjectionExec: expr=[MIN(t.a) FILTER (WHERE t.a > Int64(1))@0 as x]
> 02)--AggregateExec: mode=Single, gby=[], aggr=[MIN(t.a) FILTER (WHERE t.a > Int64(1))]
> 03)----MemoryExec: partitions=1, partition_sizes=[1]
> 
> statement ok
> drop table t;
> ```

This test is for the original issue https://github.com/apache/datafusion/issues/10878

Reply:
> > I'm not sure what is this test for 🤔
> > Is it possible to move to slt?
> > ```
> > statement ok
> > set datafusion.sql_parser.dialect = 'Postgres';
> > 
> > statement ok
> > create table t (a float) as values (1), (2), (3);
> > 
> > query TT
> > explain select min(a) filter (where a > 1) as x from t;
> > ----
> > logical_plan
> > 01)Projection: MIN(t.a) FILTER (WHERE t.a > Int64(1)) AS x
> > 02)--Aggregate: groupBy=[[]], aggr=[[MIN(t.a) FILTER (WHERE t.a > Float32(1)) AS MIN(t.a) FILTER (WHERE t.a > Int64(1))]]
> > 03)----TableScan: t projection=[a]
> > physical_plan
> > 01)ProjectionExec: expr=[MIN(t.a) FILTER (WHERE t.a > Int64(1))@0 as x]
> > 02)--AggregateExec: mode=Single, gby=[], aggr=[MIN(t.a) FILTER (WHERE t.a > Int64(1))]
> > 03)----MemoryExec: partitions=1, partition_sizes=[1]
> > 
> > statement ok
> > drop table t;
> > ```
> 
> This test is for the original issue #10878

I guess the test you write only test if the value is correct or not, didn't test if the name is expected